src/Appwrite/Event/Build.php

```diff
@@ -10,12 +10,26 @@ class Build extends Event
     protected string $type = '';
     protected ?Document $resource = null;
     protected ?Document $deployment = null;
+    protected ?Document $template = null;
 
     public function __construct()
     {
         parent::__construct(Event::BUILDS_QUEUE_NAME, Event::BUILDS_CLASS_NAME);
     }
 
+    /**
+     * Sets template for the build event.
+     *
+     * @param Document $template
+     * @return self
+     */
+    public function setTemplate(Document $template): self
+    {
+        $this->template = $template;
+
+        return $this;
+    }
+
```

Comment:
Why do we need this ? Can we not get this from the deployment/build document  ?

Reply:
If there is templte, we clone it, merge with their repo and commit&push.
This should only run once. If it was on deployment/build, it would re-run with every deployment.